File: app/schemas/wallet.py
from datetime import datetime

from pydantic import BaseModel, ConfigDict


class WalletResponse(BaseModel):
    id: int
    user_id: int
    balance: int
    created_at: datetime
    updated_at: datetime

    model_config = ConfigDict(from_attributes=True)


class TransactionResponse(BaseModel):
    id: int
    amount: int
    transaction_type: str
    description: str | None
    created_at: datetime

    model_config = ConfigDict(from_attributes=True)

File: frontend/src/types/wallet.ts
export type Wallet = {
  id: number;
  user_id: number;
  balance: number;
  created_at: string;
  updated_at: string;
};

export type Transaction = {
  id: number;
  amount: number;
  transaction_type: string;
  description: string | null;
  created_at: string;
};

export type WalletTopUpRequest = {
  amount: number;
};

export type WalletTopUpResponse = {
  message: string;
  wallet_balance: number;
  credited_amount: number;
};
